temperature/crud.py

```diff
@@ -0,0 +1,65 @@
+import datetime
+
+from fastapi import HTTPException, Depends, status
+from fastapi.responses import JSONResponse
+from sqlalchemy.orm import Session
+from sqlalchemy.future import select
+from sqlalchemy.ext.asyncio import AsyncSession
+from sqlalchemy.exc import IntegrityError
+from websockets.legacy import async_timeout
+
+from temperature import utils
+from weather import models
+from weather.crud import get_all_cities
+from dependencies import get_db
+
+
+async def get_temperature_list(db: AsyncSession):
+    result = await db.execute(
+        select(models.Temperature)
+    )
+    return result.scalars()
+
+
+async def get_temperatures_by_city_id(db: AsyncSession, city_id: int):
+    result = await db.execute(
+        select(models.Temperature).filter(models.Temperature.city_id == city_id)
+    )
+    temperature = result.scalars().first()
+
+    return temperature
+
+
+async def set_temperature(db: AsyncSession):
+    try:
+        await utils.check_service_connection()
```

Comment:
The `check_service_connection` function is called to verify service availability. Ensure that this function is correctly implemented in the `utils` module and handles exceptions appropriately.